foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/MachineLearningResourcesGenerator.java

```diff
@@ -0,0 +1,33 @@
+package com.boozallen.aiops.mda.generator;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::MDA
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aiops.mda.generator.common.PipelineEnum;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.fermenter.mda.generator.GenerationContext;
+
+public class MachineLearningResourcesGenerator extends DataLineageResourcesGenerator {
```

Comment:
I like the idea to break these out into separate generator classes. However, I think it makes more sense to have an `AbstractLineageResourcesGenerator` that contains the `generate` function and then your datalineage and modellineage child classes can set their own ` private PipelineEnum pipelineType`